flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -542,6 +579,13 @@ public void onBuffer(Buffer buffer, int sequenceNumber, int backlog) throws IOEx
 			if (backlog >= 0) {
 				onSenderBacklog(backlog);
 			}
+
+			if (notifyReceivedBarrier != null) {
+				receivedCheckpointId = notifyReceivedBarrier.getId();
```

Comment:
`RemoteInputChannel#onBuffer` is only called by the same netty thread. Because `receivedCheckpointId` is only updated and read from this method, it is thread safe.